packages/runtime/src/runtimes/react/components/PreviewProvider.tsx

```diff
@@ -1,55 +1,30 @@
 'use client'
 
-import { ReactNode, useEffect, useMemo } from 'react'
+import { type PropsWithChildren, useEffect, useMemo } from 'react'
 
-import { ReactRuntime } from '../react-runtime'
-import { StoreContext } from '../hooks/use-store'
 import * as ReactBuilderPreview from '../../../state/react-builder-preview'
-import * as ReactPage from '../../../state/react-page'
-import { MakeswiftHostApiClient } from '../../../api/react'
-import { registerDocumentEffect } from '../../../state/actions'
-import { useReactRuntime } from '../../../next/context/react-runtime'
-import { MakeswiftHostApiClientProvider } from '../../../next/context/makeswift-host-api-client'
 
-type Props = {
-  client: MakeswiftHostApiClient
-  rootElements?: Map<string, ReactPage.Element>
-  children?: ReactNode
-}
+import { useReactRuntime } from '../hooks/use-react-runtime'
+import { StoreContext } from '../hooks/use-store'
+import { useMakeswiftHostApiClient } from '../host-api-client'
+import { ReactRuntime } from '../react-runtime'
 
-export default function PreviewProvider({ client, children, rootElements }: Props): JSX.Element {
+export default function PreviewProvider({ children }: PropsWithChildren): JSX.Element {
   const runtime = useReactRuntime()
+  const client = useMakeswiftHostApiClient()
   const store = useMemo(
     () =>
       ReactBuilderPreview.configureStore({
         preloadedState: runtime ? runtime.store.getState() : ReactRuntime.store.getState(),
-        rootElements,
         client,
       }),
-    [client, rootElements, runtime],
+    [client, runtime],
   )
 
   useEffect(() => {
     store.setup()
     return () => store.teardown()
   }, [store])
 
-  useEffect(() => {
-    const unregisterDocuments = Array.from(rootElements?.entries() ?? []).map(
-      ([documentKey, rootElement]) =>
-        store.dispatch(registerDocumentEffect(ReactPage.createDocument(documentKey, rootElement))),
-    )
-
-    return () => {
-      unregisterDocuments.forEach(unregisterDocument => {
-        unregisterDocument()
-      })
-    }
-  }, [store, rootElements])
```

Comment:
Moved to [`useRegisterDocuments` effect](https://github.com/makeswift/makeswift/pull/836/files#diff-574436562edd329137174ba80e6ecdb186f78f8e2b7e6cbc929afe8ede60e4e9R33)